docs/mocksupport.rst

```diff
@@ -0,0 +1,131 @@
+.. _`Mock support`:
+
+Mock support
+============
+
+The pywbem PyPI package provides unit testing support for its users via its
+``pywbem_mock`` Python package. This package allows users of pywbem to
+define FakedWBEMConnections and CIM objects (classes, qualifier declarations,
+and instances) that will be processed by these connections. In effect this
+creates a fake WBEM Server with limited capabilities.
+
+The faked HMC environment is set up by creating an instance of the
+:class:`pywebm_mock.FakeWBEMConnection` class instead of the
+:class:`pywbem.WBEMConnection` class:
+
+The fake WBEMConnection is set up by creating an instance of the
+:class:`pywbem_mock.WBEMConnection` class instead of the
+:class:`pywbem_mock.WBEMConnection` class:
+
+.. code-block:: python
+
+    import pywbem
+    import pywbem_mock
+
+    conn = pywbem_mock.FakedWBEMConnection()
+
+    ... Add server CIM object resources (in this case some classes)
+
+    classes = conn.EnumerateClasses();
+    . . .
+
+Generally this mock environment supports:
+
+1. All of the WBEMConnection request methods that communicate with the WBEM
+server.
+
+2. Multiple namespaces.
+
+3. TODO
+
+It does this by mocking WBEM Server responses at the common level below the
+WBEMConnection methods such as GetClass, etc. but before the requests are
+formatted into HTTP.  This allows a mock repository to built by the user
+into which the data is inserted for any gived test.
+
+It does NOT support:
+
+1. CIM/XML protocol security
+2. Dynamic WBEM Server providers
+3. It does not support processing of the queries defined for ExecQuery in
+languages like CQL and WQL.  The mocker just returns found instances.
+4. It does not support the filter processing of the FQL filter query language
+   used by the Open... operations.  It returns the same data as if the filter
+   did not exist.
+5. It does not support time delay for operations.  The responses are returned
+   as soon as the mocker processes the data in the repository.
+
+.. _`Adding Data to the Fake Repository`:
+
+Adding Data to the Fake Repository
+----------------------------------
+
+There are three methods in the fake WBEMConnection to add data to the
+mock repository
+
+1. Add pywbem CIM Objects directly to the repository.
+This is the method FakedWBEMConnection.add_cimobjects().
+
+2. Compile pywbem objects from strings with the pywbem MOF compiler from strings.
+This is the method FakedWBEMConnection.compile_mof_str().
+
+3. Compile pywbem objects into the fake repository from files.
+This is the method FakedWBEMConnection.compile_mof().
+
+
+.. _`FakedWBEMConnection.add_cimobjects()`:
```

Comment:
Maybe the list of methods in the previous text is sufficient here, and we should leave the actual description of the methods to their docstrings.

Reply:
TODO. I think you are right but want to read it all over again

Reply:
Now shown in list of discussion points, below.